### ISSUE TITLE
row widget with crossAxisAlignment of baseline fails to render in Web CanvasKit

### ISSUE THREAD

**Issue by halvorsen** [flutter/flutter#74102]
row widget with crossAxisAlignment of baseline fails to render in Web CanvasKit. One instance it caused the widget and surrounding widgets to not appear, in another instance I had the widget render at an incorrect position in the stack.

Flutter (Channel master, 1.26.0-2.0.pre.366, on macOS 11.1 20C69 darwin-x64, locale en-US)

**Comment by iapicca** [comment]
> row widget with crossAxisAlignment of baseline fails to render in Web CanvasKit. One instance it caused the widget and surrounding widgets to not appear, in another instance I had the widget render at an incorrect position in the stack.
> 
> Flutter (Channel master, 1.26.0-2.0.pre.366, on macOS 11.1 20C69 darwin-x64, locale en-US)

@halvorsen 
if I understand correctly you are using a [Row](https://api.flutter.dev/flutter/widgets/Row-class.html) with  `CrossAxisAlignment.baseline` in a [Stack](https://api.flutter.dev/flutter/widgets/Stack-class.html)


<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(
    MaterialApp(
      home: Material(
        child: Stack(
          children: [
            const Align(
              alignment: Alignment.centerLeft,
              child: Text('left'),
            ),
            Align(
              alignment: Alignment.center,
              child: Row(
                textBaseline: TextBaseline.ideographic,
                crossAxisAlignment: CrossAxisAlignment.baseline,
                children: [
                  for (var i = 0; i < 5; i++) const FlutterLogo(size: 40)
                ],
              ),
            ),
            const Align(
              alignment: Alignment.centerRight,
              child: Text('right'),
            ),
          ],
        ),
      ),
    ),
  );
}

```

</details>
<details>
<summary>screenshots</summary>

[flutter run -d chrome](https://user-images.githubusercontent.com/45637986/104808893-cc39b180-57f1-11eb-93c4-70087b7ce7a0.png)

[flutter run -d chrome --dart-define=FLUTTER_WEB_USE_SKIA=true -v
](https://user-images.githubusercontent.com/45637986/104808986-95b06680-57f2-11eb-9173-90a637574dd8.png)

</details>
<details>
<summary>logs</summary>

```bash
[ +183 ms] Launching lib/main.dart on Chrome in debug mode...
[ +103 ms] Updating assets
[  +83 ms] Waiting for connection from debug service on Chrome...
[  +89 ms] <- reset
[   +8 ms] /home/francesco/snap/flutter/common/flutter/bin/cache/dart-sdk/bin/dart --disable-dart-dev
/home/francesco/snap/flutter/common/flutter/bin/cache/artifacts/engine/linux-x64/frontend_server.dart.snapshot --sdk-root /home/francesco/snap/flutter/common/flutter/bin/cache/flutter_web_sdk/
--incremental --target=dartdevc --debugger-module-names --experimental-emit-debug-metadata -DFLUTTER_WEB_AUTO_DETECT=true --output-dill /tmp/flutter_tools.TMTARL/flutter_tool.ZVFHWH/app.dill
--libraries-spec file:///home/francesco/snap/flutter/common/flutter/bin/cache/flutter_web_sdk/libraries.json --packages /home/francesco/projects/issue/.dart_tool/package_config.json -Ddart.vm.profile=false
-Ddart.vm.product=false --enable-asserts --track-widget-creation --filesystem-root /tmp/flutter_tools.TMTARL/flutter_tools.BFHOZA --filesystem-scheme org-dartlang-app --initialize-from-dill
build/c7922d95bf4a2462b75c84a97c312edb.cache.dill.track.dill --platform file:///home/francesco/snap/flutter/common/flutter/bin/cache/flutter_web_sdk/kernel/flutter_ddc_sdk.dill --no-sound-null-safety
[  +18 ms] <- compile org-dartlang-app:/web_entrypoint.dart
[+20376 ms] Waiting for connection from debug service on Chrome... (completed in 20.5s)
[        ] Synced 28.8MB.
[        ] <- accept
[        ] Caching compiled dill
[ +297 ms] Using Chromium 87.0.4280.141 snap

[ +287 ms] [CHROME]:
[        ] [CHROME]:(chrome:1421854): Gtk-WARNING **: 12:00:15.412: Theme parsing error: gtk.css:1555:23: 'font-feature-settings' is not a valid property name
[   +3 ms] [CHROME]:
[        ] [CHROME]:(chrome:1421854): Gtk-WARNING **: 12:00:15.417: Theme parsing error: gtk.css:3591:25: 'font-feature-settings' is not a valid property name
[   +1 ms] [CHROME]:
[        ] [CHROME]:(chrome:1421854): Gtk-WARNING **: 12:00:15.418: Theme parsing error: gtk.css:4053:23: 'font-feature-settings' is not a valid property name
[ +119 ms] [CHROME]:
[        ] [CHROME]:DevTools listening on ws://127.0.0.1:36375/devtools/browser/558981e1-1a86-4ca8-ba52-0d661a493bc8
[ +928 ms] DwdsInjector: Received request for entrypoint at http://localhost:38885/main_module.bootstrap.js
[   +9 ms] MetadataProvider: Loading debug metadata...
[  +24 ms] MetadataProvider: Loaded debug metadata
[  +15 ms] DwdsInjector: Injected debugging metadata for entrypoint at http://localhost:38885/main_module.bootstrap.js
[+3145 ms] DevHandler: Debug service listening on ws://127.0.0.1:36839/C_lU7ngt8aU=/ws

[  +10 ms] Debug service listening on ws://127.0.0.1:36839/C_lU7ngt8aU=/ws
[        ] Running with unsound null safety
[        ] For more information see https://dart.dev/null-safety/unsound-null-safety
[   +2 ms] Warning: Flutter's support for web development is not stable yet and hasn't
[        ] been thoroughly tested in production environments.
[        ] For more information see https://flutter.dev/web
[        ] 🔥  To hot restart changes while running, press "r" or "R".
[   +1 ms] For a more detailed help message, press "h". To quit, press "q".


```

</details>

<details>
<summary>doctor</summary>

```console
[✓] Flutter (Channel master, 1.26.0-2.0.pre.396, on Linux, locale en_US.UTF-8)
    • Flutter version 1.26.0-2.0.pre.396 at /home/francesco/snap/flutter/common/flutter
    • Framework revision 319d3a511c (4 hours ago), 2021-01-16 01:14:04 -0500
    • Engine revision a243f4df07
    • Dart version 2.12.0 (build 2.12.0-229.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /home/francesco/Android/Sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_SDK_ROOT = /home/francesco/Android/Sdk
    • Java binary at: /usr/lib/jvm/java-1.8.0-openjdk-amd64/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_275-8u275-b01-0ubuntu1~20.04-b01)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /snap/bin/chromium

[✓] Linux toolchain - develop for Linux desktop
    • clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
    • cmake version 3.10.2
    • ninja version 1.8.2
    • pkg-config version 0.29.1

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed instructions).

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Linux
    • Chrome (web)    • chrome • web-javascript • Chromium 87.0.4280.141 snap

! Doctor found issues in 1 category.

```

</details>

I don't experience any difference in the behavior w/ `canvaskit`
if the code sample I provided does not represent the situation you are describing 
please provide a **minimal reproducible code sample** together with your `flutter doctor -v` and any relevant log

**Comment by darshankawar** [comment]
@halvorsen 
Please provide a reproducible code sample that shows the issue as requested above.
Thanks.

**Comment by halvorsen** [comment]
Sorry for the late response, US was on holiday yesterday... While isolating this to give you code samples and screenshots I've realized this is my error and not Flutter's and I prematurely filed an issue. Here is what I've found. 1. This was not CanvasKit, it was a switch from stable channel to master channel (so that I could try CanvasKit), that broke my app in the locations I was using CrossAxisAlignment.basline without a textBaseline != null. 2. I was running in release mode in the master branch, as soon as I switched to debug mode I was presented with the correct warning: "crossAxisAlignment != CrossAxisAlignment.basline || textBaseline != null. @iapicca and @darshankawar thanks for the quick response.
